File: scripts/generate_hype_bets.py
import time
import random
import json
import os

USERNAMES = [
    "CryptoWhale", "Anon_77", "VIP_Diamond", "LuckBox", "SatoshiFan",
    "MumbaiKing", "DelhiBull", "GoaHighRoller", "GoldDigger", "ApexPredator",
    "RajaBetting", "JackpotGuru", "SlotWizard", "CrorepathiPro", "TokenLord",
    "DeltaWhale", "AlphaVIP", "GigaChancer", "Ruler777", "HighStakeHustler"
]

GAMES = [
    "Sweet Bonanza", "Gates of Olympus", "Book of Dead", "The Dog House",
    "Crash", "Limbo", "Plinko", "Mines", "Dice", "Aviator", "Crazy Time",
    "Lightning Roulette", "Infinite Blackjack", "Flappy Chicken", "Krunker 3D"
]

DATA_DIR = "data"
FILE_PATH = os.path.join(DATA_DIR, "hype_bets.json")

def generate_bet():
    username = random.choice(USERNAMES)
    game = random.choice(GAMES)
    
    # 75% Win rate to create "high grade hype"
    is_win = random.random() < 0.75
    
    if is_win:
        # High value wagers
        bet_amount = random.choice([10000, 20000, 25000, 50000, 100000, 150000, 200000, 250000, 500000])
        # High-hype multipliers ranging from 1.5x up to massive 500x
        multiplier = random.choice([1.5, 2.0, 2.5, 3.0, 5.0, 8.0, 10.0, 15.0, 20.0, 25.0, 50.0, 75.0, 100.0, 150.0, 250.0, 500.0])
        payout = int(bet_amount * multiplier)
    else:
        bet_amount = random.choice([5000, 10000, 20000, 50000])
        multiplier = 0.0
        payout = 0

    # Color coding based on multipliers (standard emerald, amber, neon-purple, slate-500)
    if multiplier == 0.0:
        color = "text-slate-500"
    elif multiplier < 5.0:
        color = "text-emerald-400"
    elif multiplier < 20.0:
        color = "text-amber-400"
    else:
        color = "text-neon-purple animate-pulse"

    return {
        "user": username,
        "bet": f"₹{bet_amount:,}",
        "mult": f"{multiplier}x",
        "win": f"₹{payout:,}",
        "raw_bet": bet_amount,
        "raw_payout": payout,
        "raw_mult": multiplier,
        "game": game,
        "color": color
    }

def main():
    if not os.path.exists(DATA_DIR):
        os.makedirs(DATA_DIR)
        
    print("Starting Hype Bet Generator Workflow...")
    
    # Initial session stats
    total_wagered = 4200000  # ₹4.2M
    max_win = 850000        # ₹850K
    active_players = 1402
    
    # Seed exactly the user-requested initial bets mapped to real games
    bets = [
        {
            "user": "CryptoWhale",
            "bet": "₹50,000",
            "mult": "2.5x",
            "win": "₹125,000",
            "raw_bet": 50000,
            "raw_payout": 125000,
            "raw_mult": 2.5,
            "game": "Sweet Bonanza",
            "color": "text-emerald-400"
        },
        {
            "user": "Anon_77",
            "bet": "₹10,000",
            "mult": "10.0x",
            "win": "₹100,000",
            "raw_bet": 10000,
            "raw_payout": 100000,
            "raw_mult": 10.0,
            "game": "Crash",
            "color": "text-amber-400"
        },
        {
            "user": "SatoshiFan",
            "bet": "₹5,000",
            "mult": "0.0x",
            "win": "₹0",
            "raw_bet": 5000,
            "raw_payout": 0,
            "raw_mult": 0.0,
            "game": "Plinko",
            "color": "text-slate-500"
        },
        {
            "user": "VIP_Diamond",
            "bet": "₹250,000",
            "mult": "1.5x",
            "win": "₹375,000",
            "raw_bet": 250000,
            "raw_payout": 375000,
            "raw_mult": 1.5,
            "game": "Gates of Olympus",
            "color": "text-emerald-400"
        },
        {
            "user": "LuckBox",
            "bet": "₹2,000",
            "mult": "50.0x",
            "win": "₹100,000",
            "raw_bet": 2000,
            "raw_payout": 100000,
            "raw_mult": 50.0,
            "game": "Limbo",
            "color": "text-neon-purple animate-pulse"
        }
    ]
    
    # Save the initial seed
    payload = {
        "bets": bets,
        "stats": {
            "totalWagered": "₹4.20M",
            "maxWin": "₹850K",
            "activePlayers": "1,402"
        }
    }
    with open(FILE_PATH, "w") as f:
        json.dump(payload, f, indent=2)
        
    while True:
        # Generate new bet
        new_bet = generate_bet()
        bets.insert(0, new_bet)
        bets = bets[:5]  # Keep last 5 wagers
        
        # Update session stats
        total_wagered += new_bet["raw_bet"]
        if new_bet["raw_payout"] > max_win:
            max_win = new_bet["raw_payout"]
        active_players = random.randint(1380, 1495)
        
        # Format stats
        total_wagered_str = f"₹{total_wagered/1000000:.2f}M" if total_wagered >= 1000000 else f"₹{total_wagered:,}"
        max_win_str = f"₹{max_win/1000:.0f}K" if max_win < 1000000 else f"₹{max_win/1000000:.2f}M"
        
        payload = {
            "bets": bets,
            "stats": {
                "totalWagered": total_wagered_str,
                "maxWin": max_win_str,
                "activePlayers": f"{active_players:,}"
            }
        }
        
        # Save to file
        try:
            with open(FILE_PATH, "w") as f:
                json.dump(payload, f, indent=2)
            # Replace ₹ with INR for console prints to avoid charmap encode errors on Windows
            console_bet = new_bet['bet'].replace('₹', 'INR ')
            console_win = new_bet['win'].replace('₹', 'INR ')
            print(f"Generated hype bet for {new_bet['user']}: bet={console_bet} mult={new_bet['mult']} win={console_win} game={new_bet['game']}")
        except Exception as e:
            print(f"Error writing to file: {e}")
            
        # Sleep for a random interval between 2 to 4 seconds
        time.sleep(random.uniform(2.0, 4.0))

if __name__ == "__main__":
    main()

File: scripts/generate_hype_bets.js
const fs = require('fs');
const path = require('path');

const USERNAMES = [
    "CryptoWhale", "Anon_77", "VIP_Diamond", "LuckBox", "SatoshiFan",
    "MumbaiKing", "DelhiBull", "GoaHighRoller", "GoldDigger", "ApexPredator",
    "RajaBetting", "JackpotGuru", "SlotWizard", "CrorepathiPro", "TokenLord",
    "DeltaWhale", "AlphaVIP", "GigaChancer", "Ruler777", "HighStakeHustler"
];

const CLOUD_GAMES = [
  "Cyberpunk 2077", "Elden Ring", "Black Myth: Wukong", 
  "Spider-Man 2", "Grand Theft Auto V", "Red Dead Redemption 2", 
  "Hogwarts Legacy", "Baldur's Gate 3", "Forza Horizon 5", 
  "Civilization VI"
];

const CASINO_GAMES = [
  "Crash", "Limbo", "Plinko", "Mines", "Dice", "Keno", 
  "Chicken Game", "Sweet Bonanza", "Gates of Olympus", "Crazy Time"
];

const DATA_DIR = path.join(__dirname, "..", "data");
const FILE_PATH = path.join(DATA_DIR, "hype_bets.json");

function generateFeedItem() {
  const username = USERNAMES[Math.floor(Math.random() * USERNAMES.length)];
  const isRental = Math.random() > 0.5;

  if (isRental) {
    const game = CLOUD_GAMES[Math.floor(Math.random() * CLOUD_GAMES.length)];
    const rates = [199, 299, 399, 499, 599, 799]; // Premium High Rates
    const hourly_rate = rates[Math.floor(Math.random() * rates.length)];
    const duration = Math.floor(Math.random() * 8) + 1; // 1 to 8 hours
    const total_cost = hourly_rate * duration;

    let color = "text-slate-500";
    if (duration > 6) {
      color = "text-neon-purple animate-pulse";
    } else if (duration > 3) {
      color = "text-amber-400";
    } else {
      color = "text-emerald-400";
    }

    return {
      user: username,
      bet: `${duration} hrs`,
      mult: `₹${hourly_rate}/hr`,
      win: `₹${total_cost.toLocaleString()}`,
      raw_bet: hourly_rate,
      raw_payout: total_cost,
      raw_mult: Number(duration),
      game: game,
      color: color,
      type: "rental"
    };
  } else {
    const game = CASINO_GAMES[Math.floor(Math.random() * CASINO_GAMES.length)];
    const amount = Math.floor(Math.random() * 49900) + 100; // Custom Wager Amount
    const won = Math.random() < 0.45;
    const multiplier = won ? parseFloat((Math.random() * 15 + 1.2).toFixed(2)) : 0.0;
    const payout = won ? Math.round(amount * multiplier) : 0;

    let color = "text-slate-500";
    if (multiplier > 10.0) {
      color = "text-neon-purple animate-pulse";
    } else if (multiplier > 3.0) {
      color = "text-amber-400";
    } else if (multiplier > 0.0) {
      color = "text-emerald-400";
    }

    return {
      user: username,
      bet: `₹${amount.toLocaleString()}`,
      mult: `${multiplier}x`,
      win: `₹${payout.toLocaleString()}`,
      raw_bet: amount,
      raw_payout: payout,
      raw_mult: Number(multiplier),
      game: game,
      color: color,
      type: "bet"
    };
  }
}

function main() {
  if (!fs.existsSync(DATA_DIR)) {
    fs.mkdirSync(DATA_DIR, { recursive: true });
  }
  
  console.log("Starting Dual-Mode activity generator (1 minute interval)...");
  
  let total_rented_hours = 14802;
  let max_duration = 24;
  let active_streams = 842;
  
  let bets = [];
  // Seed with 15 initial random items
  for (let i = 0; i < 15; i++) {
    bets.push(generateFeedItem());
  }
  
  const payload = {
    bets: bets,
    stats: {
      totalWagered: `₹${total_rented_hours}M`,
      maxWin: `₹${max_duration}M`,
      activePlayers: `${active_streams}`
    }
  };
  const tempInitialPath = `${FILE_PATH}.tmp`;
  fs.writeFileSync(tempInitialPath, JSON.stringify(payload, null, 2));
  fs.renameSync(tempInitialPath, FILE_PATH);

  const SYSTEM_CONFIG_PATH = path.join(__dirname, "..", "data", "system_config.json");
  
  function runLoop() {
    const newItem = generateFeedItem();
    bets.unshift(newItem);
    bets = bets.slice(0, 15);
    
    if (newItem.type === 'rental') {
      total_rented_hours += newItem.raw_mult;
    } else {
      total_rented_hours += Math.round(newItem.raw_bet / 1000);
    }
    
    if (newItem.type === 'rental' && newItem.raw_mult > max_duration) {
      max_duration = newItem.raw_mult;
    } else if (newItem.type === 'bet' && newItem.raw_mult > max_duration) {
      max_duration = Math.round(newItem.raw_mult);
    }
    
    active_streams = Math.floor(Math.random() * (895 - 810 + 1)) + 810;
    
    const updatePayload = {
      bets: bets,
      stats: {
        totalWagered: `₹${total_rented_hours}M`,
        maxWin: `₹${max_duration}M`,
        activePlayers: `${active_streams}`
      }
    };
    
    try {
      const tempPath = `${FILE_PATH}.tmp`;
      fs.writeFileSync(tempPath, JSON.stringify(updatePayload, null, 2));
      fs.renameSync(tempPath, FILE_PATH);
      console.log(`Generated item for ${newItem.user}: type=${newItem.type} game=${newItem.game} wager=${newItem.bet} payout=${newItem.win}`);
    } catch (e) {
      console.error(`Error writing to file: ${e}`);
    }
    
    // Read next interval from system config
    let nextInterval = 30000; // default to 30 seconds
    try {
      if (fs.existsSync(SYSTEM_CONFIG_PATH)) {
        const sysConfig = JSON.parse(fs.readFileSync(SYSTEM_CONFIG_PATH, 'utf-8'));
        if (typeof sysConfig.strategyFrequency === 'number') {
          nextInterval = sysConfig.strategyFrequency * 1000;
        }
      }
    } catch (err) {
      console.error("Failed to read system_config.json for strategyFrequency:", err);
    }
    
    setTimeout(runLoop, nextInterval);
  }

  // Start the loop with initial strategyFrequency
  let initialInterval = 30000;
  try {
    if (fs.existsSync(SYSTEM_CONFIG_PATH)) {
      const sysConfig = JSON.parse(fs.readFileSync(SYSTEM_CONFIG_PATH, 'utf-8'));
      if (typeof sysConfig.strategyFrequency === 'number') {
        initialInterval = sysConfig.strategyFrequency * 1000;
      }
    }
  } catch (e) {}
  
  console.log(`Starting dynamic activity generator (initial interval: ${initialInterval}ms)...`);
  setTimeout(runLoop, initialInterval);
}

main();
